hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/StreamDataChannel.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.container.common.transport.server.ratis;
+
+import org.apache.ratis.proto.RaftProtos;
+import org.apache.ratis.statemachine.StateMachine;
+
+import java.io.IOException;
+
+/**
+ * For write state machine data.
+ */
+public interface StreamDataChannel extends StateMachine.DataChannel {
+
+  default void link(RaftProtos.LogEntryProto entry) throws IOException {
+  }
```

Comment:
Checked the #2860 .  This link(..) method is for 
```
    blockManager.putBlock(kvContainer, blockData);
```
We should call ContainerStateMachine.submitTask(..) instead of blockManager.putBlock(..).  Otherwise, it will skip all the steps in-between.

So, we should remove this method and also this the interface since the interface becomes empty.

Reply:
@szetszwo 
https://issues.apache.org/jira/browse/HDDS-6137 
We will add the putBlock into stream and need to get the BCSID through the link method.

Reply:
> I need to get the BCSID at the putBlock stage.

In ContainerStateMachine.link(..), build a DispatcherContext with the BCSID and then pass the DispatcherContext  to submitTask(..), which will eventually call  blockManager.putBlock(..).

Reply:
@szetszwo 
I add ​putBlock into the channel, please check the logic of smallFileChannel, blockData is temporarily stored in the channel, so I need to return blockData in the channel, Is this bad for channel abstraction.

I added a comment in #2860, please see:
https://github.com/apache/ozone/pull/2860/files#r800366861
and
https://github.com/apache/ozone/pull/2860/files#r800366033